core/point/point_test.go

```diff
@@ -17,121 +18,168 @@ func TestCreate(t *testing.T) {
 	}
 }
 
+// The below function TestPointToString() was authored by ilmanzo
 func TestPointToString(t *testing.T) {
 	result := fmt.Sprintf("%v", New(2, 3))
 	const expected = "{2,3}"
 	if result != expected {
-		t.Errorf("Point() to string: expected %v, got %v", expected, result)
+		t.Errorf("Point() to string: expected %v, got %v", expected,
+			result)
 	}
 }
 
 func TestPointToSGFTranslate(t *testing.T) {
 	// First test translation from integer-point to SGF-string-point
 	testToSGFCases := []struct {
-		desc string
-		in   *Point
-		want string
+		desc         string
+		in           *Point
+		want         string
+		expErrSubstr string
 	}{
 		{
 			desc: "Point => SGF",
-			in: &Point{
-				x: 8,
-				y: 16,
-			},
+			in:   New(8, 16),
 			want: "iq",
 		},
 		{
 			desc: "Point => SGF",
-			in: &Point{
-				x: 12,
-				y: 5,
-			},
+			in:   New(12, 5),
 			want: "mf",
 		},
 		{
 			desc: "Point => SGF",
-			in: &Point{
-				x: 33,
-				y: 8,
-			},
+			in:   New(33, 8),
 			want: "Hi",
 		},
 		{
 			desc: "Point => SGF",
-			in: &Point{
-				x: 40,
-				y: 51,
-			},
+			in:   New(40, 51),
 			want: "OZ",
 		},
+		// Error Cases:
+		// Negative X Y values
+		{
+			desc:         "Error: Point => SGF: X value < 0",
+			in:           New(-33, 8),
+			expErrSubstr: "Point X value < 0, out of range",
+		},
+		{
+			desc:         "Error: Point => SGF: Y value < 0",
+			in:           New(33, -8),
+			expErrSubstr: "Point Y value < 0, out of range",
+		},
+		// Out of range X Y values
+		{
+			desc:         "Error: Point => SGF: X value > 51",
+			in:           New(53, 8),
+			expErrSubstr: "Point X value > 51, out of range",
+		},
+		{
+			desc:         "Error: Point => SGF: Y value > 51",
+			in:           New(33, 53),
+			expErrSubstr: "Point Y value > 51, out of range",
+		},
 	}
 
 	for _, tc := range testToSGFCases {
 		t.Run(tc.desc, func(t *testing.T) {
-			toSGFOut, _ := New(tc.in.x, tc.in.y).ToSGF()
-			if toSGFOut != tc.want {
+			sgfOut, err := New(tc.in.x, tc.in.y).ToSGF()
+			cerr := errcheck.CheckCases(err, tc.expErrSubstr)
+			if sgfOut != tc.want {
```

Comment:
this is too indented. To avoid excessive indenting, break in case of error
```
if blah {
  t.Errorf(...)
  return // break
}
// continue on
```